st3/sublime_lib/region_manager.py

```diff
@@ -0,0 +1,61 @@
+import sublime
+
+from ._compat.typing import Optional, List, TypeVar
+from .flags import RegionOption
+
+
+__all__ = ['RegionManager']
+
+
+T = TypeVar('T')
+
+
+def coalesce(*values: Optional[T]) -> T:
+    return next(value for value in values if value is not None)
+
+
+class RegionManager:
+    def __init__(
+        self,
+        view: sublime.View,
+        key: Optional[str] = None,
+        *,
+        scope: Optional[str] = None,
+        icon: Optional[str] = None,
+        flags: Optional[RegionOption] = None
+    ):
+        self.view = view
+
+        if key is None:
+            self.key = str(id(self))
+        else:
+            self.key = key
+
+        self.scope = scope
+        self.icon = icon
+        self.flags = flags
+
+    def __del__(self) -> None:
+        self.erase()
+
+    def set(
+        self,
+        regions: List[sublime.Region],
+        *,
+        scope: Optional[str] = None,
```

Comment:
Should the scope/icon/flags overrides of this function affect the manager's defaults? If not, what use case do we imagine for manually overriding these for a single `set`?

Reply:
The `_coalesce` arguments were backward; the arguments to `set` are supposed to supersede the defaults. I've fixed it.

Reply:
Ah sure, I didn't even catch this. I mean the existance of the overrides in this context in general.

Reply:
I don't have a particular use case in mind. Even without those arguments to `set`, you could always just change the attributes manually. I'll ping rwols in the issue to see if he has an opinion.